### PR TITLE
[CPU] Split node migration on nGraph.

### PR DESCRIPTION
**Pull request by nshchego**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp [hunk 1]
```diff
@@ -279,28 +279,26 @@ void MKLDNNGraph::Replicate(const CNNNetwork &network, const MKLDNNExtensionMana
             graphEdges.push_back(edge);
         }
 
-//        for (auto &out_data : layer->outData) {
-//            if (getInputTo(out_data).empty()) {
-//                unused_data.insert(out_data);
-//            }
-//        }
+        if (op->get_type_info() != ngraph::op::v0::Result::type_info) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp:282]: Do we really need this condition? I would expect Result::get_output_size() will always return 0

File: inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp [hunk 2]
```diff
@@ -279,28 +279,26 @@ void MKLDNNGraph::Replicate(const CNNNetwork &network, const MKLDNNExtensionMana
...
+                    unusedOutputs.push_back(op->output(oi));
+                }
+            }
+        }
+    }
+
+    // Add stub output node for unused outputs
+    for (auto unusedOutput : unusedOutputs) {
+        auto parentNode = op2node[unusedOutput.get_node_shared_ptr()];
+        auto newResult = std::make_shared<ngraph::op::v0::Result>(unusedOutput.get_node()->output(unusedOutput.get_index()));
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp:294]: Maybe I missing smt but isn't `unusedOutput.get_node()->output(unusedOutput.get_index())` == `unusedOutput`?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp [hunk 3]
```diff
@@ -56,178 +56,191 @@ static TensorDesc makeChannelBlockedTensorDesc(const Precision& precision, const
...
+bool MKLDNNSplitNode::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        auto splitOp = ngraph::as_type_ptr<const ngraph::op::v1::Split>(op);
+        if (!splitOp) {
+            errorMessage = "Node is not an instance of the Split operation.";
+            return false;
+        }
+        auto axisOp = ngraph::as_type_ptr<ngraph::op::v0::Constant>(op->get_input_node_shared_ptr(1));
+        if (!axisOp) {
+            errorMessage = "Constant expected as the second input.";
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:68]: I think `second` -> `axis` will sound more clear.

File: inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp (hunk 1 above)
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp:282]: Actually it returns 1.

File: inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp (hunk 2 above)
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp]: Fixed

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp (hunk 3 above)
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp]: Fixed

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**